lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -405,12 +403,15 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     {last_address, _} = DB.get_last_chain_address(genesis_address)
 
     with {^last_address, _} <- DB.get_last_chain_address(genesis_address, last_scheduling_date),
-         {:ok, _, _} <-
-           NodeRenewal.decode_transaction_content(content),
+         {:ok, _, _} <- NodeRenewal.decode_transaction_content(content),
          true <-
-           Enum.all?(
-             Map.keys(authorized_keys),
-             &Utils.key_in_node_list?(nodes, &1)
+           authorized_keys
```

Comment:
I think it will be more clear without the pipeline by extracting variables and do simple comparison.